### PR TITLE
Have CSI driver use 2.12 Lustre client

### PR DESCRIPTION
**Pull request by jacobwolfaws**
**Is this a bug fix or adding new feature?**
update client
**What is this PR about? / Why do we need it?**
This PR moves the csi driver to use the 2.12 lustre client
**What testing is done?**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ RUN OS=$TARGETOS ARCH=$TARGETARCH make $TARGETOS/$TARGETARCH
 FROM amazonlinux:2 AS linux-amazon
 RUN yum update -y
 RUN yum install util-linux libyaml -y \
-    && amazon-linux-extras install -y lustre2.10
+    && amazon-linux-extras install -y lustre
```
Comment (l1n) [inline Dockerfile:25]: Hey, question, why aren't we version pinning here anymore?

Comment (khoang98) [reply Dockerfile:25]: We chose to move to a versionless repo to easier facilitate upgrades for customers. This helps protect against the scenario where customer pins an older version in some configuration file and forgets about it.

Comment (l1n) [reply Dockerfile:25]: This seems like it risks issues with unintentional upgrades. Please, reconsider offering a pinned version here (or at least tag with the built Dockerfile version in the official ECR images)

**Pull request merged**